dev/com.ibm.ws.cdi.internal/src/com/ibm/ws/cdi/internal/archive/liberty/CDIArchiveImpl.java

```diff
@@ -274,57 +272,58 @@ public ExtendedModuleInfo getModuleInfo() throws CDIException {
 
     private ClientModuleInfo getClientModuleInfo() throws CDIException {
         ModuleInfo moduleInfo = getModuleInfo(); // throws CDIException
-        if ( !(moduleInfo instanceof ClientModuleInfo) ) {
+        if (!(moduleInfo instanceof ClientModuleInfo)) {
             return null;
         } else {
             return (ClientModuleInfo) moduleInfo;
         }
     }
-    
+
     public ModuleMetaData getModuleMetaData() throws CDIException {
         ExtendedModuleInfo useModuleInfo = getModuleInfo();
-        return ( (useModuleInfo == null) ? null : useModuleInfo.getMetaData() );
+        return ((useModuleInfo == null) ? null : useModuleInfo.getMetaData());
     }
 
     @Override
     public MetaData getMetaData() throws CDIException {
-        if ( isModule() ) {
+        if (isModule()) {
             return getModuleMetaData();
-        } else {
+        } else if (application != null) {
             return application.getApplicationMetaData();
         }
```

Comment:
The Javadoc for this method indicates that it must not be called for non-application archives.

Looking at the rest of the class, `application` should only be `null` for non-application archives, right?

Maybe an `OperationNotSupportedException` if `application` is `null` would be more appropriate?

Reply:
It should be safe, the old behaviour was an NPE so we know nobody relies on it.

Reply:
I think that should be safe. Assuming all non-application archives are also not modules they would NPE before so an exception now won't break anything.

Reply:
Looking at containerInfo I can't make that assumption without digging into it. But I can put the new exception where the old NPE was.